src/hydro_forces.cpp

```diff
@@ -353,58 +353,48 @@ double TestHydro::setVelHistory(double val, int step, int b_num, int index) {
 *******************************************************************************/
 std::vector<double> TestHydro::ComputeForceHydrostatics() {
 	std::vector<double> position, displacement;
-	unsigned total_dofs = 6 * num_bodies;
-	position.resize(total_dofs, 0.0);
-	displacement.resize(total_dofs, 0.0);
+	assert(num_bodies > 0);
 
-	// orientation initialized to system current pos/rot vectors
 	for (int b = 0; b < num_bodies; b++) {
-		for (int i = 0; i < 3; i++) {
-			unsigned b_offset = 6 * b;
-			if (b_offset + i + 3 > total_dofs || b_offset + i < 0) {
-				std::cout << "temp index in hydrostatic force is bad " << std::endl;
-			}
-			position[i + b_offset] = bodies[b]->GetPos()[i];
-			position[i + 3 + b_offset] = bodies[b]->GetRot().Q_to_Euler123()[i]; // TODO check if we want this, or GetWvel_par()?
+		// initialize variables
+		auto body = bodies[b];
+		auto body_h5file = file_info[b];
+		auto rho = body_h5file.rho;
+		unsigned b_offset = 6 * b;  // force_hydrostatic has 6 elements for each body so to skip to the next body we move 6 spaces
+		auto body_force_hydrostatic = &force_hydrostatic[b_offset];
+		auto gg = body->GetSystem()->Get_G_acc().Length();
+
+		// hydrostatic stiffness due to offset from equilibrium
+		auto body_position = body->GetPos();
+		auto body_rotation = body->GetRot().Q_to_Euler123();
+		// calculate displacement
+		chrono::ChVectorN<double, 6> displacement;
+		for (int ii = 0; ii < 3; ii++) {
+			displacement[ii] = body_position[ii] - equilibrium[ii];
+			displacement[ii + 3] = body_rotation[ii] - equilibrium[ii + 3];
```

Comment:
I think the equilibrium vector is initialized in the constructor and is a 6N dimensional vector. It includes equilibrium information for each body, do when calculating the displacement here we should use body_offset again. Otherwise this only ever uses the equilibrium for the first body to calculate the displacement for each body.
```suggestion
			displacement[ii] = body_position[ii] - equilibrium[ii + body_offset];
			displacement[ii + 3] = body_rotation[ii] - equilibrium[ii + 3 + body_offset];
```

Reply:
That's right, `equilibrium` should be a body-level variables, I declared and used a `body_equilibrium` variable to use here (and also renamed `displacement` as `body_displacement` for clarity)